eclipse-theia/theia
# electron: Removed `disallowReloadKeybinding` option, fixes Issue #11099

#### What it does

The application configuration option `disallowReloadKeybinding` is no longer used, therefore I removed that config option from the following file: 

`theia/dev-packages/application-package/src/application-props.ts` 

And its mentioning in the test file: 

`theia/packages/core/src/browser/frontend-application-config-provider.spec.ts`

#### Fixes #11099 

#### How to test
There is nothing much to test after this small change although I have run `npm test` which finished as expected.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
